kof-operator/internal/controller/remote-secret/remote_secret_manager.go

```diff
@@ -0,0 +1,156 @@
+package remotesecret
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	kcmv1alpha1 "github.com/K0rdent/kcm/api/v1alpha1"
+	"github.com/go-logr/logr"
+	istio "github.com/k0rdent/kof/kof-operator/internal/controller/isito"
+	"istio.io/istio/istioctl/pkg/multicluster"
+	"istio.io/istio/pkg/kube"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type RemoteSecretManager struct {
+	client client.Client
+	IIstioRemoteSecretCreator
+}
+
+const (
+	IstioRoleLabelKey = "k0rdent.mirantis.com/istio-role"
+	IstioRoleChild    = "child"
+)
+
+func New(c client.Client) *RemoteSecretManager {
+	return &RemoteSecretManager{
+		client:                    c,
+		IIstioRemoteSecretCreator: NewIstioRemoteSecret(),
+	}
+}
+
+// Function handles the creation of a remote secret
+func (rs *RemoteSecretManager) Create(clusterDeployment *kcmv1alpha1.ClusterDeployment, ctx context.Context, request ctrl.Request) error {
+	log := log.FromContext(ctx)
+
+	if !rs.isClusterDeploymentReady(*clusterDeployment.GetConditions()) {
+		return nil
+	}
+
+	if !rs.hasIstioChildRoleLabel(clusterDeployment.Labels) {
+		return nil
+	}
+
+	kubeconfig, err := rs.GetKubeconfigFromSecret(log, ctx, request)
+	if err != nil {
+		return err
+	}
+
+	remoteSecret, err := rs.CreateRemoteSecret(kubeconfig, log, request.Name)
+	if err != nil {
+		return err
+	}
+
+	remoteSecret.Namespace = request.Namespace
+	if err := rs.client.Create(ctx, remoteSecret); err != nil {
+		log.Error(err, "failed to create secret on mothership cluster")
+		return err
+	}
+
+	return nil
+}
+
+// Function retrieves and decodes a kubeconfig from a Secret
+func (rs *RemoteSecretManager) GetKubeconfigFromSecret(logger logr.Logger, ctx context.Context, request ctrl.Request) ([]byte, error) {
```

Comment:
Do we need to pass `logger` if it is part of `ctx`?
```suggestion
func (rs *RemoteSecretManager) GetKubeconfigFromSecret(ctx context.Context, request ctrl.Request) ([]byte, error) {
	logger := log.FromContext(ctx)
```